crates/ra_syntax/src/ast/expr_extensions.rs

```diff
@@ -239,16 +239,34 @@ impl ast::Literal {
     pub fn kind(&self) -> LiteralKind {
         match self.token().kind() {
             INT_NUMBER => {
-                let allowed_suffix_list = [
+                let int_suffix_list = [
                     "isize", "i128", "i64", "i32", "i16", "i8", "usize", "u128", "u64", "u32",
                     "u16", "u8",
                 ];
+
+                // The lexer treats e.g. `1f64` as an integer literal. See
+                // https://github.com/rust-analyzer/rust-analyzer/issues/1592
+                // and the comments on the linked PR.
+                let float_suffix_list = [
```

Comment:
Is this the best place to handle this? It appears roughly analogous to where the special case is handled in rustc itself - see https://github.com/rust-lang/rust/blob/a7f28678bbf4e16893bb6a718e427504167a9494/src/libsyntax/parse/literal.rs#L424

Reply:
I think this is indeed the right place to do this. `1` is indeed, *lexically*, an integer literal